Examples/Swift/ViewController.swift

```diff
@@ -14,23 +16,54 @@ enum ExampleMode {
     case multipleWaypoints
 }
 
-class ViewController: UIViewController, MGLMapViewDelegate, CLLocationManagerDelegate, NavigationMapViewDelegate {
+class ViewController: UIViewController, MGLMapViewDelegate, CLLocationManagerDelegate {
     
+    //MARK: - IBOutlets
+    @IBOutlet weak var mapView: NavigationMapView!
+    @IBOutlet weak var longPressHintView: UIView!
+    
+    @IBOutlet weak var simulationButton: UIButton!
+    @IBOutlet weak var startButton: UIButton!
+    
+    @IBOutlet weak var clearMap: UIButton!
+    
+    //MARK: Properties
     var waypoints: [Waypoint] = []
     var currentRoute: Route? {
+        get {
+            return routes?.first
+        }
+        set {
+            self.startButton.isEnabled = (newValue != nil)
```

Comment:
nit: drop self ×4

Reply:
Agree. Willfix.

Reply:
As it turns out, the last two are required, but the first two are redundant and have been removed.